code/modules/customitems/item_spawning.dm

```diff
@@ -1,236 +1,276 @@
+	else
+		//If we dont have a db, we need to load the json file
+		load_from_file = 1
+
+	if(load_from_file)
+		log_debug("Custom Items: Loading from File")
+		//Check if the json file exists
+		if(fexists("config/custom_items.json"))
+			log_debug("Custom Items: Loading from json")
+			var/list/loaded_items = list()
+			var/item_id = 0
+			try
+				loaded_items = json_decode(return_file_text("config/custom_items.json"))
+			catch(var/exception/e)
+				log_debug("Custom Items: Failed to load custom_items.json: [e]")
+
+			for(var/item in loaded_items)
+				//TODO: Check for existance of the vars first
+				item_id += 1
+				var/datum/custom_item/ci = new()
+				ci.id = item_id
+				ci.usr_ckey = item["ckey"]
+				ci.usr_charname = item["character_name"]
+				ci.item_path = text2path(item["item_path"])
+				ci.item_data["name"] = item["item_name"]
+				ci.item_data["desc"] = item["item_desc"]
+				ci.additional_data = item["additional_data"]
+				custom_items.Add(ci)
+			log_debug("Custom Items: Loaded [length(custom_items)] custom items")
+		else if(fexists("config/custom_items.txt")) //TODO: Retire that at some point down the line
+			log_debug("Custom Items: Loading from txt")
+			log_and_message_admins("The deprecated custom_items.txt file is used. Migrate to SQL or JSON.")
+			//If we dont have the json file, we might have the old file so lets try that
+			var/datum/custom_item/current_data
+			for(var/line in text2list(file2text("config/custom_items.txt"), "\n"))
+				line = trim(line)
+				if(line == "" || !line || findtext(line, "#", 1, 2))
+					continue
+
+				if(findtext(line, "{", 1, 2) || findtext(line, "}", 1, 2)) // New block!
+					if(current_data && current_data.usr_ckey && current_data.usr_charname)
+						custom_items.Add(current_data)
+					current_data = null
+
+				var/split = findtext(line,":")
+				if(!split)
+					continue
+				var/field = trim(copytext(line,1,split))
+				var/field_data = trim(copytext(line,(split+1)))
+				if(!field || !field_data)
+					continue
+
+				if(!current_data)
+					current_data = new()
+
+				switch(field)
+					if("ckey")
+						current_data.usr_ckey = ckey(field_data)
+					if("character_name")
+						current_data.usr_charname = lowertext(field_data)
+					if("item_path")
+						current_data.item_path = text2path(field_data)
+					if("item_name")
+						current_data.item_data["name"] = field_data
+					if("item_icon")
+						continue
+					if("inherit_inhands")
+						continue
+					if("item_desc")
+						current_data.item_data["desc"] = field_data
+					if("req_access")
+						current_data.req_access = text2num(field_data)
+					if("req_titles")
+						current_data.req_titles = text2list(field_data,", ")
+					if("kit_name")
+						continue
+					if("kit_desc")
+						continue
+					if("kit_icon")
+						continue
+					if("additional_data")
+						current_data.additional_data = field_data
+	if(load_from_file == 2) //insert the item into the db
```

Comment:
```suggestion
	if(load_from_file == 2 && custom_items.len) //insert the item into the db
```

I forget if this is the correct syntax but not doing this will spam log_debug every time you run an empty server with an empty DB.